dlt/destinations/job_client_impl.py

```diff
@@ -645,11 +648,15 @@ def _row_to_schema_info(self, query: str, *args: Any) -> StorageSchemaInfo:
 
     def _delete_schema_in_storage(self, schema: Schema) -> None:
         """
-        Delete all stored versions with the same name as given schema
+        Delete all stored versions with the same name as given schema.
+        Fails silently if versions table does not exist
         """
-        name = self.sql_client.make_qualified_table_name(self.schema.version_table_name)
-        (c_schema_name,) = self._norm_and_escape_columns("schema_name")
-        self.sql_client.execute_sql(f"DELETE FROM {name} WHERE {c_schema_name} = %s;", schema.name)
+        with suppress(DatabaseUndefinedRelation):
```

Comment:
this will not work when running inside a transaction (if supports_ddl_transaction is active), because the exception is raise when the transaction context exits. We could check wether the table exists before we try to delete from it. DELETE FROM IF EXISTS does not seem to be standardized in sql, so we'd probably need a couple of different implementations for each destination.

Reply:
so we still need this "suppress"? checking for initialized dataset should be enough. OFC user can drop the table schema manually, then it will fail. 
my take: remove "suppress" and run tests again. you can restore the test client to the old form, in most cases we have transactions and as you say - this does not work anyway